jupyter/nmm-cache.ipynb

```diff
@@ -140,12 +146,45 @@
     "del read"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "1e0a1e14",
+   "metadata": {},
+   "source": [
+    "Die Methode `clean()` prüft ob der Cache seine maximale Größe überschritten hat. Ist dies der Fall wird das minimale Limit um eins erhöht und alle Einträge, deren Limit geringer ist, werden aus dem Cache entfernt."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e3e4e408",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def clean(self):\n",
+    "    while len(self.cache) > self.max_size:\n",
+    "        self.min_limit += 1\n",
+    "        pre_len = len(self.cache)\n",
+    "        to_delete = []\n",
+    "        for key,value in self.cache.items():\n",
+    "            limit = int(str(key[8]),16)\n",
+    "            if limit < self.min_limit:\n",
+    "                to_delete.append(key)\n",
+    "        for key in to_delete:\n",
+    "            self.cache.pop(key)\n",
```

Comment:
FYI, `pop()` gibt den Wert an der Stelle `key` zurück. In diesem Fall würde folgendes verwendet werden:
```py
del self.cache[key]
```

Reply:
Changed